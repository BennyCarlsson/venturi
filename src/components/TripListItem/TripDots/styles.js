import styled from 'styled-components'

export const TripDotsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`
export const Dot = styled.div`
  width: 1px;
  height: 7px;
  margin-top: 2px;
  margin-bottom: 2px;
  background-color: #719fb0;
  border-radius: 1px;
`
export const LetterCircle = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 50%;
  background-color: #719fb0;
  height: 24px;
  width: 24px;
  box-shadow: -1px 4px 7px 2px rgb(0 0 0 / 31%);
  font-size: 18px;
  font-weight: 400;
  color: #ffffff;
`
