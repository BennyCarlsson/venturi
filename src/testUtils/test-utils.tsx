import { configureStore } from '@reduxjs/toolkit';
import locationsReducer from '../redux/locationsSlice';
import testReducer from '../redux/testSlice';
import tripReducer from '../redux/tripSlice';
import dialogReducer from '../redux/dialogSlice';
import { render as rtlRender, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ReactElement } from 'react';
import { RootState } from 'redux/store';

type RenderOptionsType = { initialState?: Partial<RootState> } & Omit<RenderOptions, 'queries'>;
function render(ui: ReactElement, { initialState = {}, ...renderOptions }: RenderOptionsType) {
  const store = configureStore({
    reducer: {
      test: testReducer,
      trip: tripReducer,
      locations: locationsReducer,
      dialog: dialogReducer
    },
    preloadedState: initialState
  });
  function Wrapper({ children }: any) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render };
