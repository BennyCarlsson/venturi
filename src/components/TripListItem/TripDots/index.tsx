import { Dot, LetterCircle, TripDotsWrapper } from './styles'

const TripDots = () => {
  return (
    <TripDotsWrapper>
      <LetterCircle>A</LetterCircle>
      <Dot />
      <Dot />
      <Dot />
      <Dot />
      <LetterCircle>B</LetterCircle>
    </TripDotsWrapper>
  )
}

export default TripDots
