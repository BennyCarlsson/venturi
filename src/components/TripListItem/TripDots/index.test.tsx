import { render, screen } from 'testUtils/test-utils'
import TripDots from '.'

describe('<TripDots />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<TripDots />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render without crashing', () => {
    render(<TripDots />, {})
    expect(screen).toBeTruthy()
  })
})
