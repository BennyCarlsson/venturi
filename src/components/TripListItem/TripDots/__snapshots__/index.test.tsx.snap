// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TripDots /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ jVeENm"
  >
    <div
      class="sc-dlfnbm gREHbW"
    >
      A
    </div>
    <div
      class="sc-gsTCUz dosEAT"
    />
    <div
      class="sc-gsTCUz dosEAT"
    />
    <div
      class="sc-gsTCUz dosEAT"
    />
    <div
      class="sc-gsTCUz dosEAT"
    />
    <div
      class="sc-dlfnbm gREHbW"
    >
      B
    </div>
  </div>
</DocumentFragment>
`;
